tornado-api/src/main/java/uk/ac/manchester/tornado/api/TornadoBackend.java

```diff
@@ -23,24 +23,91 @@
 import uk.ac.manchester.tornado.api.enums.TornadoDeviceType;
 import uk.ac.manchester.tornado.api.enums.TornadoVMBackendType;
 
+/**
+ * Backend Interfaces in TornadoVM: It contains the minimum amount of methods to be query per backend,
+ * such as the total number of devices, query the default device, etc.
+ */
 public interface TornadoBackend {
 
+    /**
+     * Returns the default device accelerator. If TornadoVM has access to multiple accelerators,
+     * it will select one as default one. The default accelerator depends on the backends installed.
+     * TornadoVM prioritizes compute capabilities and memory available to sort the list of devices.
+     * The default device is the first in this internal list.
+     * 
+     * @return {@link TornadoDevice}
+     */
     TornadoDevice getDefaultDevice();
 
+    /**
+     * Sets the default device. The default device is set in the first position of an internal linked list.
+     * This method swaps the selected device to be used as default.
+     *
+     * @param index
+     *     Device Index within the list of devices per backend.
+     */
     void setDefaultDevice(int index);
 
-    int getBackendCounter();
+    /**
+     * Returns the total number of devices per backend. TornadoVM guarantees at least one device.
```

Comment:
are we sure about the guarantee? if there is no device suitable for a backend, how will TornadoVM guarantee to return one device?

Reply:
Because if there are no devices, no backend will be instantiated. The error when there are no backends will be thrown early in the Backend exploration. Once it finds a backend, then TornadoVM will instantiate these classes.

Reply:
ok, thanks @jjfumero